### DISCUSSION TITLE
Lost data upgrading to 2.x: Cannot read all data

### DISCUSSION THREAD

**Discussion by DanielHaitink** [plausible/analytics#3491] [Self-Hosted Support] [answered]
I have updated my plausible docker compose container and I am trying to migrate the data. However this results in an error: 

```
Running create-domains-lookup
  -> Query: CREATE TABLE IF NOT EXISTS domains_lookup(site_id UInt64, domain String) ENGINE Memory;

Execute? [Y/n]:
    create-domains-lookup Done!

------------------------------------------------------------------------------
Populate domains-lookup with postgres sites [Y/n]:
** (Ch.Error) Code: 33, e.displayText() = DB::Exception: Cannot read all data. Bytes read: 27. Bytes expected: 100.: (at row 2)
 (version 21.3.2.5 (official build))

    lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
    lib/ecto/adapters/clickhouse/schema.ex:35: Ecto.Adapters.ClickHouse.Schema.insert_all/8
    lib/ecto/repo/schema.ex:59: Ecto.Repo.Schema.do_insert_all/7
    lib/plausible/data_migration/numeric_ids.ex:124: Plausible.DataMigration.NumericIDs.run/1
    nofile:1: (file)
    (stdlib 4.2) erl_eval.erl:748: :erl_eval.do_apply/7
```

Would someone have any idea how to fix this problem?

**Comment by ruslandoga** [comment]
👋 @DanielHaitink 

It seems like some domains fail to be [inserted](https://github.com/plausible/analytics/blob/f977351ae2ba4d79f432978317a49e64dcaafa89/lib/plausible/data_migration/numeric_ids.ex#L125-L130) into the `domains_lookup` table.

**Reply by ruslandoga** [reply]
Can you please connect to the running `plausible` instance and try executing that failing step row by row? It might give us the idea for why it is failing.

```console
$ cd hosting
$ docker compose exec plausible bin/plausible remote
```
```elixir
iex> import Ecto.Query
iex> mappings = Plausible.Site |> select([s], %{site_id: s.id, domain: s.domain}) |> Plausible.Repo.all()
iex> mappings # are there any special symbols in domain names, etc?
iex> Plausible.IngestRepo.insert_all(Plausible.DataMigration.NumericIDs.DomainsLookup, mappings)
```

**Reply by ruslandoga** [reply]
And don't worry. It doesn't seem like any data has been lost yet :)

**Reply by DanielHaitink** [reply]
No special characters or something like that. At the time of migrating one of the websites (id = 5) was empty and never obtained data. Could this be a problem?

**Reply by ruslandoga** [reply]
Hm. I don't seem to be getting any errors when inserting these values

**Reply by ruslandoga** [reply]
I think it might be due to your ClickHouse server version being a bit old. Mine is `23.3.7.5` and yours seems to be `21.3.2.5`.

**Reply by ruslandoga** [reply, marked as answer]
Maybe you could try upgrading to `22.6`? The v2 guide shows how in the beginning: https://github.com/plausible/analytics/discussions/3132 (under "Ensure you are using a new ClickHouse version")

**Reply by DanielHaitink** [reply]
Thank you! That fixed the problem for me!
